PassportOfficeOpenAPI.yaml

```diff
@@ -97,137 +132,167 @@ paths:
-      - name: number
-        in: path
-        description: Passport number
-        required: true
-        schema:
-          type: string
-          format: dddddd
-          description: 6 digits
-          example: 123456
-    get:
+  /passport:
+    post:
       tags:
         - Passport
-      summary: Returns one passport
-      operationId: getPassportBySeriesNumber
+      summary: Add one new passport
+      operationId: postPassport
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Passport'
       responses:
         '200':
-          $ref: '#/components/responses/ReturnPassport'
-        '401':
-          description: Authorisation required
-        '404':
-          description: Not found
+          description: OK entity created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Passport'
+        '400':
+          description: Bad Request - invalid syntax
+        '422':
+          $ref: '#/components/responses/InvalidData'
+
     patch:
       tags:
         - Passport
       summary: Edit passport
       operationId: editPassport
+      parameters:
+        - name: passportID
+          in: query
+          required: true
+          description: Old passport ID
+          schema:
+            type: string
       requestBody:
-        $ref: '#/components/requestBodies/PostPassport'
-      responses:
-        '200':
-          description: OK, accepted
-        '409':
-          description: Data conflict
-        '401':
-          description: Authorisation required
-        '404':
-          description: Not found
-        '452': #Custom error
-          description: Bad Data
-    delete:
-      tags:
-        - Passport
-      summary: Mark passport as deleted by series and number
-      operationId: deletePassport
+        required: true
+        content:
+          application/json:
+            schema:
+              oneOf:
+                - $ref: '#/components/schemas/RFPassport'
+                - $ref: '#/components/schemas/ForeignRFPassport'
+                - $ref: '#/components/schemas/NonRFPassport'
+              discriminator:
+                propertyName: passportType
       responses:
         '200':
-          description: OK, accepted
-        '401':
-          description: Authorisation required
+          description: OK, entity updated
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Passport'
         '404':
           description: Not found
-  /passport:
-    post:
+        '400':
+          description: Bad Request - invalid syntax
+        '422':
+          $ref: '#/components/responses/InvalidData'
+
+    put:
       tags:
         - Passport
-      summary: add one passport
-      operationId: postPassport
+      summary: Add new passport to swap the old one
```

Comment:
That concerns - PUT is usually used for updating a whole entity, returning the same entity with the same ID. This descripion assumes another entity gets created.
PUT is also an idempotent operation, so should not produce any other entities.